src/tronls.jl

```diff
@@ -64,32 +66,90 @@ stats = tron(nls)
 # output
 "Execution stats: first-order stationary"
 ```
+
+```jldoctest; output = false
+using JSOSolvers, ADNLPModels
+F(x) = [x[1] - 1.0; 10 * (x[2] - x[1]^2)]
+x0 = [-1.2; 1.0]
+nls = ADNLSModel(F, x0, 2, zeros(2), 0.5 * ones(2))
+solver = TronSolverNLS(nls)
+stats = solve!(solver, nls)
+# output
+"Execution stats: first-order stationary"
+```
 """
-function tron(
+mutable struct TronSolverNLS{T, V <: AbstractVector{T}, Op <: AbstractLinearOperator{T}} <: AbstractOptimizationSolver
+  x::V
+  xc::V
+  gx::V
+  gt::V
+  gpx::V
+  Hs::V
+  H::Op
+  tr::TrustRegion{T, V}
+  Fc::V
+  Av::V
+  Atv::V
+end 
+
+function TronSolverNLS(nlp::AbstractNLSModel{T, V};) where {T, V <: AbstractVector{T}}
+  nvar = nlp.meta.nvar
+  nequ = nlp.nls_meta.nequ
+  x = V(undef, nvar)
+  xc = V(undef, nvar)
+  gx = V(undef, nvar)
+  gt = V(undef, nvar)
+  gpx = V(undef, nvar)
+  Hs = V(undef, nvar)
+  H = hess_op!(nlp, x, Hs)
```

Comment:
```suggestion
  H = hess_op(nlp, x)
```

Reply:
I think we will allocate every time we compute a product even when using `mul!` https://github.com/JuliaSmoothOptimizers/NLPModels.jl/blob/d6c1a4375eca62a6cf769ca31bf0a54dc5f0f6cb/src/nlp/api.jl#L1181